src/openfermion/ops/_fermion_operator.py

```diff
@@ -205,3 +205,75 @@ def is_molecular_term(self):
             if not (particles == spin == 0):
                 return False
         return True
+
+    def freeze_orbitals(self, occupied, unoccupied=[], prune=True):
+        """Fix some orbitals to be occupied and others unoccupied.
+
+        Removes all operators acting on the specified orbitals, and flips the
+        sign of each term as appropriate. Terms whose expectation value
+        between states with the specified occupancy is zero are removed.
+        Additionally, the indices are renumbered such that the frozen orbitals
+        are eliminated.
+
+        Note that the sign change accounts for both the reordering of operators
+        needed to group the operators acting on the frozen orbitals, as well
+        as the impact of pruning frozen orbitals on the Jordan-Wigner strings
+        of operators acting on unfrozen orbitals.
+
+        Args:
+        occupied: A list containing the indices of the orbitals that are to be
+            assumed to be occupied.
+        unoccupied: A list containing the indices of the orbitals that are to
+            be assumed to be unoccupied.
+        """
+
+        frozen = [(index, 1) for index in occupied] + \
+                 [(index, 0) for index in unoccupied]
+
+        # Loop over each orbital to be frozen. Within each term, move all
+        # ops acting on that orbital to the right side of the term, keeping
+        # track of sign flips that come from swapping operators.
+        for item in frozen:
+            new_terms = {}
+            for term in self.terms:
+                new_term = []
+                new_coef = self.terms[term]
+                current_occupancy = item[1]
+                n_ops = 0  # Number of operations on index that have been moved
+                n_swaps = 0  # Number of swaps that have been done
+
+                for op in enumerate(reversed(term)):
```

Comment:
This seems like it borrows a lot from typical normal ordering, which we already have routines for in this file.  Do you think it's possible to use that routine or modify that routine in a way that you can use it here as a core element in your function to minimize code duplication?

Reply:
It's not obvious to me how to use normal_ordered to simplify this part of the code. I want to bring all of the ladder operators corresponding to the given orbital to the right side of the term, whereas normal_ordered will put them somewhere in the middle.